api/classes/publisher.js

```diff
@@ -1,411 +1,412 @@
+      if (!files.length) {
+        return;
+      }
 
-    return Promise.map(files, function(file) {
-      return PublishedFileQueries.createOne({
-        file_id: file.id,
-        published_id: publishedProject.id,
-        path: file.path,
-        buffer: file.buffer
-      })
-      .then(function() {
-        return upload(fileRoot + file.path, file.buffer, remixData);
-      })
-      .then(success('CREATE', username))
-      .catch(failure('CREATE', username));
+      return Promise.map(files, file => {
+        return publishedFilesQueryBuilder.createOne({
+          file_id: file.id,
+          published_id: this.publishedProject.id,
+          path: file.path,
+          buffer: file.buffer
+        })
+        .then(() => upload(
+          `${this.publishRoot}${file.path}`,
+          file.buffer,
+          this.remixData
+        ))
+        .then(success(`CREATE`, this.user.name))
+        .catch(failure(`CREATE`, this.user.name));
+      });
     });
-  });
-}
+  }
 
-function uploadModifiedFiles() {
-  var self = this;
-  var remixData = self.remixData;
-  var fileRoot = self.publishRoot;
-  var username = self.user.name;
+  uploadModifiedFiles() {
+    const remixData = this.remixData;
+    const fileRoot = this.publishRoot;
+    const username = this.user.name;
 
-  function updateModelAndUpload(publishedFile) {
-    var id = publishedFile.id;
-    delete publishedFile.id;
+    function updateModelAndUpload(publishedFile) {
+      const id = publishedFile.id;
 
-    return PublishedFileQueries
+      delete publishedFile.id;
+
+      return publishedFilesQueryBuilder
       .updateOne(id, publishedFile)
       .then(function() {
-        return upload(fileRoot + publishedFile.path, publishedFile.buffer, remixData);
+        return upload(
+          `${fileRoot}${publishedFile.path}`,
+          publishedFile.buffer,
+          remixData
+        );
       })
-      .then(success('UPDATE', username))
-      .catch(failure('UPDATE', username));
-  }
-
-  return PublishedFileQueries
-  .getAllModifiedFiles(self.publishedProject.id)
-  .then(function(publishedFiles) {
-    if (!publishedFiles.length) {
-      return;
+      .then(success(`UPDATE`, username))
+      .catch(failure(`UPDATE`, username));
     }
 
-    return Promise.map(publishedFiles, function(publishedFile) {
-      var oldPath = publishedFile.oldPath;
-      delete publishedFile.oldPath;
-
-      if (oldPath === publishedFile.path) {
-        return updateModelAndUpload(publishedFile);
+    return publishedFilesQueryBuilder
+    .getAllModifiedFiles(this.publishedProject.id)
+    .then(function(publishedFiles) {
+      if (!publishedFiles.length) {
+        return;
       }
 
-      return remove(fileRoot + oldPath)
-      .then(function() {
-        return updateModelAndUpload(publishedFile);
+      return Promise.map(publishedFiles, function(publishedFile) {
+        const oldPath = publishedFile.oldPath;
+
+        delete publishedFile.oldPath;
```

Comment:
Interesting! Yeah I think I'll set it to `null` since it will practically have the same effect.